pkg/controller/ttlafterfinished/ttlafterfinished_controller.go

```diff
@@ -116,43 +132,78 @@ func (tc *Controller) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
-func (tc *Controller) addJob(obj interface{}) {
-	job := obj.(*batch.Job)
-	klog.V(4).Infof("Adding job %s/%s", job.Namespace, job.Name)
+func (tc *Controller) addResource(obj interface{}) {
+	switch resource := obj.(type) {
+	case *batch.Job:
+		klog.V(4).Infof("Adding job %s/%s", resource.Namespace, resource.Name)
+
+		if resource.DeletionTimestamp == nil && needsCleanup(resource) {
+			tc.enqueue(resource)
+		}
+	case *v1.Pod:
+		klog.V(4).Infof("Adding pod %s/%s", resource.Namespace, resource.Name)
 
-	if job.DeletionTimestamp == nil && needsCleanup(job) {
-		tc.enqueue(job)
+		if resource.DeletionTimestamp == nil && needsCleanup(resource) {
+			tc.enqueue(resource)
+		}
 	}
 }
 
-func (tc *Controller) updateJob(old, cur interface{}) {
-	job := cur.(*batch.Job)
-	klog.V(4).Infof("Updating job %s/%s", job.Namespace, job.Name)
+func (tc *Controller) updateResource(old, cur interface{}) {
+	switch resource := cur.(type) {
+	case *batch.Job:
+		klog.V(4).Infof("Updating job %s/%s", resource.Namespace, resource.Name)
 
-	if job.DeletionTimestamp == nil && needsCleanup(job) {
-		tc.enqueue(job)
+		if resource.DeletionTimestamp == nil && needsCleanup(resource) {
+			tc.enqueue(resource)
+		}
+	case *v1.Pod:
+		klog.V(4).Infof("Updating pod %s/%s", resource.Namespace, resource.Name)
+		if resource.DeletionTimestamp == nil && needsCleanup(resource) {
+			tc.enqueue(resource)
+		}
 	}
 }
 
-func (tc *Controller) enqueue(job *batch.Job) {
-	klog.V(4).Infof("Add job %s/%s to cleanup", job.Namespace, job.Name)
-	key, err := controller.KeyFunc(job)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %#v: %v", job, err))
-		return
+func (tc *Controller) enqueue(obj interface{}) {
+	switch resource := obj.(type) {
+	case *batch.Job:
+		klog.V(4).Infof("Add job %s/%s to cleanup", resource.Namespace, resource.Name)
+		key, err := controller.KeyFunc(resource)
```

Comment:
`controller.KeyFunc` uses resource namespace + name as keys. This would have problems when the queue has resources of more than one kind. When Job and Pod has the same namespace + name, the controller won't be able to differentiate them.